### PR TITLE
Make audit recognise head-only and devel-only in all taps

### PR DESCRIPTION
**Pull request by rkrug**
- add wildcard into expressions to identify head-only and dev-only
  taps to avoid "head-only" and "devel-only" errors when recipie is in
  *head-only or *devel-only tap

### CODE REVIEW

**Comment by rkrug** [comment]
This one is working and tested locally.

**Comment by dunn** [comment]
Looks like you accidentally committed a patch file too.

**Comment by rkrug** [comment]
Envoyé de mon iPhone

> Le 2 juil. 2015 à 18:14, Alex Dunn <notifications@github.com> a écrit :
> 
> Looks like you accidentally committed a patch file too.
> 

True - sorry.

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -421,11 +421,11 @@ def audit_homepage
   end
 
   def audit_specs
-    if head_only?(formula) && formula.tap.to_s.downcase != "homebrew/homebrew-head-only"
+    if head_only?(formula) && formula.tap.to_s.downcase !~ /.*head-only/
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: `/-head-only/` should be enough here, I believe.

File: Library/Homebrew/cmd/audit.rb [hunk 2]
```diff
@@ -421,11 +421,11 @@ def audit_homepage
...
       problem "Head-only (no stable download)"
     end
 
-    if devel_only?(formula) && formula.tap.to_s.downcase != "homebrew/homebrew-devel-only"
+    if devel_only?(formula) && formula.tap.to_s.downcase !~ /.*devel-only/
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: and `/-devel-only/` here.

File: Library/Homebrew/cmd/0001-Make-audit-recognise-hed-only-and-devel-only-in-all-.patch [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+     end
+ 
+-    if devel_only?(formula) && formula.tap.to_s.downcase != "homebrew/homebrew-devel-only"
++    if devel_only?(formula) && formula.tap.to_s.downcase != "*/homebrew-devel-only"
+       problem "Devel-only (no stable download)"
+     end
+ 
+-- 
+2.4.5
+
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/0001-Make-audit-recognise-hed-only-and-devel-only-in-all-.patch]: Please remove this file.

**Comment by mikemcquaid** [comment]
@Homebrew/owners I think I prefer this approach to https://github.com/Homebrew/homebrew/issues/41413 and I think given there's a few tap authors asking for this we should merge something.

**Comment by jawshooah** [comment]
@mikemcquaid This would require any head-only or devel-only formulae to be split into separate taps, right?

**Comment by mikemcquaid** [comment]
@jawshooah Yes, they would. `head-only` provides a fairly shoddy user experience in Homebrew currently, unfortunately, and `audit` checks for that for that reason.

**Comment by jawshooah** [comment]
Makes sense. It's probably a good convention to follow anyway, so that a user can confidently `brew install` any formulae in a tap not explicitly labeled as head/devel-only.

Shall I close the other PR, or would you like to wait for input from other maintainers?

**Comment by mikemcquaid** [comment]
@jawshooah Thanks for understanding. Let's wait for input for now. Long-term I'd love to see HEAD install doing something smart using `git describe` or the Subversion revision number to calculate a `123-HEAD` version or something.

**Comment by DomT4** [comment]
A while back I threw together a rough idea of storing the `.git` or `.svn` folders for the `HEAD` clone in the formula's prefix, and then running `git pull` or similar with each `brew upgrade`, but I suspected from having a bunch of stuff locally I pull at the same time the performance impact would have sucked so I dropped it.

Theoretically we could handle the performance hit by making people specify they wanted to update/upgrade their installed `HEAD` formulae :octopus:.

**Comment by jawshooah** [comment]
Yeah, this issue has been discussed at length in #13197. The problems with `HEAD` upgrades haven't changed, unfortunately.

**Comment by rkrug** [comment]
I just committed the required changes. Let me know if you would prefer a clean pull request with only one commit.

**Pull request closed without merging**